src/v/storage/disk_log_appender.cc

```diff
@@ -50,7 +50,7 @@ ss::future<> disk_log_appender::initialize() {
 }
 
 bool disk_log_appender::segment_is_appendable(model::term_id batch_term) const {
-    if (!_seg || !_seg->has_appender()) {
+    if (!_seg || !_seg->has_appender() || _seg->is_tombstone()) {
```

Comment:
Just wondering with the write lock taken, is this needed still for the bug? (fine keeping it since it seems correct, but I'm curious)

Reply:
You are right, i think it may not be strictly required in current situation with the lock